### PR TITLE
Fix some issues around permissions on rest services

### PR DESCRIPTION
**Pull request by micedre**

### CODE REVIEW

**Comment by micedre** [comment]
Fix #245

File: sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/services/PermissionService.java [hunk 1]
```diff
@@ -59,7 +61,12 @@ public boolean isPasswordManager(String realm, String userStorage) {
 
   public boolean isApplicationManager(String realm, String userStorage, String application) {
     List<String> searchRoleList =
-        getSearchRoleList(realm, userStorage, application, passwordManagerRoleList);
+        getSearchRoleList(realm, userStorage, application, applicationManagerRoleList);
+    return checkIfUserGetRoles(searchRoleList);
+  }
+
+  public boolean isApplicationManager(String realm) {
```
Comment (CChemin) [inline sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/services/PermissionService.java:68]: maybe isApplication**s**Manager or something like that to understand that it has rights on all application on all userstorage ? (if it's what is done here)

**Review by CChemin — COMMENTED** [review]
is it possible to add documentation on the different permissions, how to enable them and the properties used ? 😇

File: sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/ApplicationController.java [hunk 2]
```diff
@@ -178,7 +178,7 @@
             description = "Application does'nt exist",
             content = {@Content(mediaType = "application/json")})
       })
-  @PreAuthorize("@NewAuthorizeMethodDecider.isWriter(#realm,#storage)")
+  @PreAuthorize("@NewAuthorizeMethodDecider.isAppManager(#realm,#name)")
```
[the remark this answers is not in the document]
Comment (micedre) [reply sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/ApplicationController.java:181]: neither, it should be #id instead, fix in progress

File: sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/services/PermissionService.java (hunk 1 above)
Comment (micedre) [reply sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/services/PermissionService.java:68]: No this methods check if the user is applicationManager for one app in the realm.

File: sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/ApplicationController.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (micedre) [reply sugoi-api-rest-services/src/main/java/fr/insee/sugoi/services/controller/ApplicationController.java]: should be fixed now

**Review by CChemin — APPROVED** [review]

**Pull request merged**